pinot-broker/src/test/java/org/apache/pinot/broker/broker/HelixBrokerStarterTest.java

```diff
@@ -18,287 +18,197 @@
-        return _helixAdmin.getResourceExternalView(getHelixClusterName(), DINING_TABLE_NAME).getPartitionSet().size()
-            == SEGMENT_COUNT;
-      }
-    }, 30000L);
-
-    externalView = _helixAdmin.getResourceExternalView(getHelixClusterName(), DINING_TABLE_NAME);
-    Assert.assertEquals(externalView.getPartitionSet().size(), SEGMENT_COUNT);
-    tableArray = brokerRoutingTableBuilderMap.keySet().toArray();
-    Arrays.sort(tableArray);
-    Assert.assertEquals(Arrays.toString(tableArray), "[coffee_OFFLINE, dining_OFFLINE, dining_REALTIME]");
-
-    // Wait up to 30s for routing table to reach the expected size
-    waitForPredicate(new Callable<Boolean>() {
-      @Override
-      public Boolean call()
-          throws Exception {
-        Map<String, List<String>> routingTable =
-            brokerRoutingTableBuilderMap.get(DINING_TABLE_NAME).getRoutingTables().get(0);
-        return routingTable.values().iterator().next().size() == SEGMENT_COUNT;
-      }
-    }, 30000L);
-
-    Assert.assertEquals(
-        brokerRoutingTableBuilderMap.get(DINING_TABLE_NAME).getRoutingTables().get(0).values().iterator().next().size(),
-        SEGMENT_COUNT);
+        .addNewSegment(OFFLINE_TABLE_NAME, SegmentMetadataMockUtils.mockSegmentMetadata(RAW_TABLE_NAME), "downloadUrl");
+
+    TestUtils.waitForCondition(
+        aVoid -> routing.getRoutingTable(routingTableLookupRequest).values().iterator().next().size()
+            == NUM_OFFLINE_SEGMENTS + 1, 30_000L, "Failed to add the new segment into the routing table");
+
+    // Add a new table with different broker tenant
+    String newRawTableName = "newTable";
+    String newOfflineTableName = TableNameBuilder.OFFLINE.tableNameWithType(newRawTableName);
+    TableConfig newTableConfig =
+        new TableConfig.Builder(TableType.OFFLINE).setTableName(newRawTableName).setBrokerTenant("testBroker").build();
+    _helixResourceManager.addTable(newTableConfig);
+
+    // Broker tenant should be overridden to DefaultTenant
+    TableConfig newTableConfigInCluster = _helixResourceManager.getTableConfig(newOfflineTableName);
+    assertNotNull(newTableConfigInCluster);
+    assertEquals(newTableConfigInCluster.getTenantConfig().getBroker(), TagNameUtils.DEFAULT_TENANT_NAME);
+
+    brokerResourceIdealState = _helixAdmin.getResourceIdealState(getHelixClusterName(), Helix.BROKER_RESOURCE_INSTANCE);
+    assertEquals(brokerResourceIdealState.getInstanceSet(newOfflineTableName).size(), NUM_BROKERS);
+
+    TestUtils.waitForCondition(aVoid -> {
+      Map<String, String> newTableStateMap =
+          _helixAdmin.getResourceExternalView(getHelixClusterName(), Helix.BROKER_RESOURCE_INSTANCE)
+              .getStateMap(newOfflineTableName);
+      return newTableStateMap != null && newTableStateMap.size() == NUM_BROKERS;
+    }, 30_000L, "Failed to find all brokers for the new table in the brokerResource ExternalView");
+
+    assertTrue(routing.routingTableExists(newOfflineTableName));
```

Comment:
We seem to be testing for both realtime and offline segments to show up in the routing table,. Since the routing tables are different, isnt it better to have both in the tests? I think you have removed the realtime segment check?

Reply:
That one is tested in line 142